ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java

```diff
@@ -0,0 +1,62 @@
+package io.syndesis.qe.pages;
+
+import static com.codeborne.selenide.Condition.visible;
+
+import org.openqa.selenium.By;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+
+public abstract class SyndesisPageObject {
+
+	abstract SelenideElement getRootElement();
+
+	abstract boolean validate();
+
+	SelenideElement getLink(String linkTitle) {
+		return this.getRootElement().find(By.linkText(linkTitle));
+	}
+
+	ElementsCollection getLinks(String linkTitle) {
+		return this.getRootElement().findAll(By.linkText(linkTitle));
+	}
+
+	void clickLink(String linkTitle) {
+		this.getLink(linkTitle).shouldBe(visible).click();	
+	}
+
+	void clickLinkRandom(String linkTitle) {
+		ElementsCollection links = this.getLinks(linkTitle);
+		int index = (int) Math.floor(Math.random() * links.size());
+		links.get(index).shouldBe(visible).click();
+	}
+
+	SelenideElement getElementByCssSelector(String cssSelector) {
+		return this.getRootElement().find(By.cssSelector(cssSelector));
+	}
+
+	SelenideElement getElementByXpath(String xpathSelector) {
+		return this.getRootElement().find(By.xpath(xpathSelector));
+	}
+
+	SelenideElement getElementByClassName(String elementClassName) {
+		return this.getRootElement().find(By.className(elementClassName));
+	}
+
+	ElementsCollection getElementsByClassName(String elementClassName) {
+		return this.getRootElement().findAll(By.className(elementClassName));
+	}
+
+	void clickElementRandom(String elementClassName) {
+		ElementsCollection elements = this.getElementsByClassName(elementClassName);
+		int index = (int) Math.floor(Math.random() * elements.size());
+		elements.get(index).shouldBe(visible).click();
+	}
+
+	void selectOption(SelenideElement selectElement, String option) {
+		ElementsCollection optionElements = selectElement.findAll(By.tagName("option"));
+		optionElements = optionElements.filter(Condition.exactText(option));
+		optionElements.first().shouldBe(visible).click();
```

Comment:
Add check if there is exactly one element in optionElements to avoid/throw error?

Reply:
Thank you and general idea?